pkg/controller/job/controller.go

```diff
@@ -403,10 +398,6 @@ func (jm *JobManager) manageJob(activePods []*api.Pod, successful, unsuccessful
 	} else if active < parallelism {
 		// how many executions are left to run
 		diff := *job.Spec.Completions - successful
-		// for RestartPolicyNever we need to count unsuccessful pods as well
```

Comment:
How job interacts with restart policy was not part of the initial proposal. We really wanted to get job in but we wanted to discuss this interaction in design review over the next couple months while it was in experimental. Thanks for pointing out that incorrect comment, but with the logic the way it was [this comment](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/experimental/types.go#L403-L404) is incorrect because both successful pods and failed pod will be counted towards my completion.

Please see #14186 which is an umbrella issue for jobcontroller enhancements.